source/scripting_v3/GTA/Entities/Vehicles/Vehicle.cs

```diff
@@ -85,6 +82,15 @@ public float DirtLevel
 			set => Function.Call(Hash.SET_VEHICLE_DIRT_LEVEL, Handle, value);
 		}
 
+		/// <summary>
+		/// Gets or sets the amount of paint fade on this <see cref="Vehicle"/> between 0.0 (new) and 1.0 (fully worn).
+		/// </summary>
+		public float PaintFadeLevel
```

Comment:
My apologies for the massive delay in responding; I was incredibly busy this week and only now was able to get time to return to this project.

What can I do to test this out? From what I can tell ENVEFF in GTAV only refers to the paint on vehicles; it causes aircraft metal to show through and become oxidized; it is what causes vehicle paint to fade away and become rust. On the DeLorean in BTTFV, we abuse it to create the icing effect on our stainless steel vehicle body.

If there is anything I can do to test this, or if it would be better named something else, please let me know and I'll get right on it! Thanks!

Reply:
I haven't found a counterexample actually. I'll let you know if there's a case that enveff can be something other than paint fade beside the DeLorean in BTTFV.

Reply:
We should rename the property to `EnvEffScale` (or `EnvEffLevel`) rather than `PaintFadeLevel`, I found a good example where enveff means something more than the paint fade. Try messing with the enveff value of A Nightmare Cerberus (cerberus3).

![271590_20230903205626_1](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/9353978/668d03fd-30c0-4ffc-9322-aff99136393b)
Zero EnvEff.

![271590_20230903205629_1](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/9353978/74bfecca-fd70-41cc-8d83-f006b1ac24d1)
Full EnvEff.

So, we can say that the enveff natives get or set how intense certain enveff diffuse texture opacity is. You see the Rhino's paint and Barracks paint "fade", but that's because the alternative texture appear more clearly. `SET_VEHICLE_ENVEFF_SCALE` sets a byte field on `CCustomShaderEffectVehicle` on `CVehicleDrawHandler` (of `CVehicle`), like `(signed int)(float)(a2 * 255.0)` (just use 0xFF if the calculated value is 0xFF or more).

You want to see another example like how the enveff scale value affects the Nightmare Cerberus? Try messing with the enveff value of A Khanjali.